src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -80,16 +97,63 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
         catch (Exception ex)
         {
-            Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
-            await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace, cancellationToken);
+            if (ex is JsonException or CmsEventException)
+            {
+                Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
+                await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace,
+                    cancellationToken);
+            }
```

Comment:
There's no return statement after deadlettering the message; wouldn't this mean that the redelivery code below would execute, and then possibly fail if it gets to line 118?